res/controllers/common-controller-scripts.js

```diff
@@ -392,10 +393,20 @@ bpm.tapButton = function(deck) {
     var now = new Date()/1000;   // Current time in seconds
     var tapDelta = now - bpm.tapTime;
     bpm.tapTime=now;
+    if (bpm.tap.length<1.0) {
+      bpm.previousTapDelta=tapDelta;
+    }
+    // assign tapDelta in cases where the button has not been pressed previously
     if (tapDelta>2.0) { // reset if longer than two seconds between taps
         bpm.tap=[];
         return;
     }
+    if ((tapDelta > bpm.previousTapDelta*1.8)||(tapDelta<bpm.previousTapDelta*0.4)) {
```

Comment:
If this condition happens and we return / ignore the tap, the next tap will only be even farther away from `previousTapDelta`, so I think what this is saying is that if your tapping ever varies by more than 1.8x or less than 0.4x, you have to wait 2 seconds to hit the `tapDelta > 2.0` reset ?

Reply:
Not quite, the tapDelta is out of the accepted range, the tap interval will be rejected but the previousTapDelta and the bpm[] doesnt change so the tapDelta of that particular tap just gets ignored.

Reply:
Oh, I see -- bpm.tapTime is updated regardless. So math-wise, this appears to be a way to make the mean less sensitive to outliers by dropping them, but the criteria for ignoring an outlier is similarity to the immediately previous tapdelta?

Reply:
Exactly. Maybe the name `previousTapdelta` is a bit missleading as it refers to the tapdelta of the previous **successful** tapdelta. When you accidentally double pressed the button, the next tapdelta will be the time between the tap and the accidental press (which is not an issue in my opinion as the double press will be very close to the actual tap time wise). it works likewise when you miss a tap.
The only fundamental flaw that this method has is that the function assumes that the first tapDelta is correct. So It won't work accurately (and reject subsequent correct taps) when either one of the first two taps is incorrect.